ballerina/xml_api.bal

```diff
@@ -0,0 +1,365 @@
+# In the XML string-record conversion, this annotation can be used to override the default XML element name using the
+# `xmldata:toXML` API and validate the overridden XML element name with record field using the `xml:fromXmlString` API.
+public annotation NameConfig Name on type, record field;
+
+# Defines the namespace of the XML element
+#
+# + prefix - The value of the prefix of the namespace
+# + uri - The value of the URI of the namespace
+public type NamespaceConfig record {|
+    string prefix?;
+    string uri;
+|};
+
+# The annotation is used to specify the namespace's prefix and URI of the XML element.
+# In the XML string-record conversion, this annotation can be used to add XML namespace using the `xmldata:toXML` API and
+# validate the the XML namespace with record in the `xml:fromXmlString` API.
+public annotation NamespaceConfig Namespace on type, record field;
+
+# The annotation is used to denote the field that is considered an attribute.
+# In the XML string-record conversion, this annotation can be used to add XML attribute using the `xmldata:toXML` API and
+# validate the XML attribute with record fields in the `xml:fromXmlString` API.
+public annotation Attribute on record field;
+
+# A Parse a xml value with projection
+#
+# + v - Source XML value 
+# + options - Options to be used for filtering in the projection
+# + t - Target type to be used for filtering in the projection
+# + return - On success, returns the given target type value, else returns an `xmldata:Error`
+public isolated function fromXmlWithType(xml v, Options options = {}, typedesc<anydata> t = <>)
+        returns t|ConversionError = @java:Method {'class: "io.ballerina.stdlib.data.xml.Native"} external;
+
+# A Parser a xml string value with projection
+#
+# + s - Source XML string value or byte[] or byte-stream
+# + options - Options to be used for filtering in the projection 
+# + t - Target type to be used for filtering in the projection
+# + return - On success, returns the given target type value, else returns an `xmldata:Error`
+public isolated function fromXmlStringWithType(string|byte[]|stream<byte[], error?> s, Options options = {}, typedesc<anydata> t = <>)
+        returns t|ConversionError = @java:Method {'class: "io.ballerina.stdlib.data.xml.Native"} external;
+
+
+# Converts a `Map` or `Record` representation to its XML representation.
+# Additionally, when converting from a record, the `xmldata:Namespace`, `xmldata:Name`, and `xmldata:Attribute`
+# annotations can be used to add `namespaces`, `name of elements`, and `attributes` to XML representation.
+#
+# + mapValue - The `Map` or `Record` representation source to be converted to XML
+# + return - XML representation of the given source if the source is
+# successfully converted or else an `xmldata:Error`
+public isolated function toXml(map<anydata> mapValue) returns xml|error {
+    if mapValue is map<xml>|map<xml[]> {
```

Comment:
Is the difference in the following intentional?

```ballerina
import ballerina/data.xmldata;
import ballerina/io;

public function main() returns error? {
    map<xml> m1 = {
        greeting: xml `Hello`
    };
    xml x1 = check xmldata:toXml(m1);
    map<xml|int> m2 = {
        greeting: xml `Hello`
    };
    xml x2 = check xmldata:toXml(m2);

    io:println(x1); // <root><greeting>Hello</greeting></root>
    io:println(x2); // <greeting>Hello</greeting>
}
```

Reply:
One reason I see is, they tried to omit the java external call for those two cases.

Reply:
@prakanth97 did we fix this?

Reply:
Fixed with this [0bd3479](https://github.com/ballerina-platform/module-ballerina-data-xmldata/pull/1/commits/0bd34799b9c4477472943ee49cf6c8bdebe2dd36). Restricted the map union cases since we did not support union cases.